src/data/list/func.lean

```diff
@@ -0,0 +1,312 @@
+/-
+Copyright (c) 2019 Seul Baek. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Seul Baek
+-/
+import data.list.defs
+import data.nat.basic
+import tactic.interactive
+
+open list list.func
+
+universes u v w
+variables {α : Type u} {β : Type v} {γ : Type w}
+
+namespace list
+namespace func
+
+variables {a : α}
+variables {as as1 as2 as3 : list α}
+
+localized "notation as ` {` m ` ↦ ` a `}` := list.func.set a as m" in list.func
```

Comment:
Why is this not done after the definition of `list.func.set`?  I presume there is an import cycle?

AFAICT the `list.func` namespace is only used by the omega tactic.  So it should be safe to move the definitions here as well.

Reply:
Moved.

Reply:
I did move the definitions for this one.